src/com/goide/inspections/GoDeferGoInspection.java

```diff
@@ -36,20 +38,58 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
       @Override
       public void visitDeferStatement(@NotNull GoDeferStatement o) {
         super.visitDeferStatement(o);
+
+        if (o.getExpression() != null && o.getExpression() instanceof GoCallExpr) {
+          GoCallExpr callExpr = (GoCallExpr)o.getExpression();
+
+          if (GoPsiImplUtil.isPanic(callExpr)) {
+            holder.registerProblem(o, "defer should not call panic() directly #loc", ProblemHighlightType.WEAK_WARNING,
+                                   new GoDeleteQuickFix("Delete statement", GoDeferStatement.class));
+            return;
+          }
+
+          if (GoPsiImplUtil.isRecover(callExpr)) {
+            holder.registerProblem(o, "defer should not call recover() directly #loc", ProblemHighlightType.WEAK_WARNING,
+                                   new GoReplaceWithCorrectDeferRecoverQuickFix());
+            return;
+          }
+        }
+
         checkExpression(o.getExpression(), "defer");
       }
 
       @Override
       public void visitGoStatement(@NotNull GoGoStatement o) {
         super.visitGoStatement(o);
+
+        if (o.getExpression() != null && o.getExpression() instanceof GoCallExpr) {
+          GoCallExpr callExpr = (GoCallExpr)o.getExpression();
+
+          if (GoPsiImplUtil.isPanic(callExpr)) {
+            holder.registerProblem(o, "go should not call panic() directly #loc", ProblemHighlightType.WEAK_WARNING,
+                                   new GoDeleteQuickFix("Delete statement", GoGoStatement.class));
+            return;
+          }
+
+          if (GoPsiImplUtil.isRecover(callExpr)) {
+            holder.registerProblem(o, "go should not call recover() directly #loc", ProblemHighlightType.WEAK_WARNING,
+                                   new GoDeleteQuickFix("Delete statement", GoGoStatement.class));
+            return;
+          }
+        }
+
         checkExpression(o.getExpression(), "go");
       }
 
       private void checkExpression(@Nullable GoExpression o, String who) {
-        if (o == null) return;
-        if ((o instanceof GoCallExpr || o instanceof GoBuiltinCallExpr) && !GoPsiImplUtil.isConversionExpression(o)) {
+        if (o == null || o instanceof GoBuiltinCallExpr) return;
```

Comment:
I would like to to have a better solution for that case, and usually people don't `defer new()`. Probably I should change it do not ignore it if the function call is for `recover()` in `defer` calls.